### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by subratac** [autogluon/autogluon#2801]
- [ x] I have checked that this bug exists on the latest stable version of AutoGluon
- [x ] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
I am using the TabularPredictor as follows:

label = 'DistantMets'
save_path = 'AutogluonModels/DM'  
metric = 'f1'
predictor = TabularPredictor(label=label, eval_metric=metric, path=save_path).fit(df_train, presets='best_quality')

Then, I use it to test the model performance on the test set:

y_test = df_test[label]  
df_test_nolab = df_test.drop(columns=[label])  
df_test_nolab.head()

predictor = TabularPredictor.load("AutogluonModels/DM")
y_pred = predictor.predict(df_test_nolab)
perf = predictor.evaluate_predictions(y_true=y_test, y_pred=y_pred, auxiliary_metrics=True, silent = True)

The output does not contain the AUC statistics (`roc_auc`)

{'f1': 0.47058823529411764,
 'accuracy': 0.5909090909090909,
 'balanced_accuracy': 0.5714285714285714,
 'mcc': 0.13977131156292244,
 'precision': 0.4444444444444444,
 'recall': 0.5}

This behavior is not because the model has < 0.5 F1, since in a different situation using the same target variable, but with a different dataset, it again does not give me the `roc_auc` metric

{'f1': 0.5882352941176471,
 'accuracy': 0.6818181818181818,
 'balanced_accuracy': 0.6696428571428572,
 'mcc': 0.33195686496194077,
 'precision': 0.5555555555555556,
 'recall': 0.625}

**Expected behavior**
Performance metric should include the ROC statistic everytime

**To Reproduce**
A minimal script to reproduce the issue. Links to Colab notebooks or similar tools are encouraged.  
If the code is too long, feel free to put it in a public gist and link it in the issue: https://gist.github.com.  
In short, we are going to copy-paste your code to run it and we expect to get the same result as you.

**Screenshots**

**Installed Versions**

Which version of AutoGluon are you are using?  

AutoGluon Version:  0.6.1b20221124
Python Version:     3.9.15
Operating System:   Darwin
Platform Machine:   x86_64
Platform Version:   Darwin Kernel Version 19.6.0: Tue Jun 21 21:18:39 PDT 2022; root:xnu-6153.141.66~1/RELEASE_X86_64

**Comment by Innixma** [comment]
You are using `predict`, which does not include probabilities. `roc_auc` requires prediction probabilities to score properly. The result will include `roc_auc` score if you instead feed it the output of `predict_proba`.
